src/mutation/RelayMutationQuery.js

```diff
@@ -210,26 +212,32 @@ const RelayMutationQuery = {
         if (!trackedEdges.length) {
           return;
         }
-        if (trackedConnection.getRangeBehaviorKey() in rangeBehaviors) {
+
+        const rangeBehaviorKey = trackedConnection.getRangeBehaviorKey();
+        const rangeBehaviorValue = rangeBehaviors[rangeBehaviorKey];
+        if (rangeBehaviorKey in rangeBehaviors && rangeBehaviorValue !== REFETCH) {
           // Include edges from all connections that exist in `rangeBehaviors`.
           // This may add duplicates, but they will eventually be flattened.
           trackedEdges.forEach(trackedEdge => {
             mutatedEdgeFields.push(...trackedEdge.getChildren());
           });
         } else {
-          // If the connection is not in `rangeBehaviors`, re-fetch it.
-          warning(
-            false,
-            'RelayMutation: The connection `%s` on the mutation field `%s` ' +
-            'that corresponds to the ID `%s` did not match any of the ' +
-            '`rangeBehaviors` specified in your RANGE_ADD config. This means ' +
-            'that the entire connection will be refetched. Configure a range ' +
-            'behavior for this mutation in order to fetch only the new edge ' +
-            'and to enable optimistic mutations.',
-            trackedConnection.getStorageKey(),
-            parentName,
-            parentID
-          );
+          // If the connection is not in `rangeBehaviors` or we have explicitly
+          // set the behavior to `refetch`, re-fetch it.
+          if (rangeBehaviorValue !== REFETCH) {
+            warning(
+              false,
```

Comment:
```
warning(
   rangeBehavior === REFETCH,
  ...
);
```

?

It's possible we're doing the `warning(false, ...)` thing all over the place to avoid the overhead of a called to a function, but this is not a hot code path and outside of `__DEV__` the function is just `emptyFunction`, so close to free...